### PR TITLE
Feature(Fb Pixel): Compatibilty with cloud mode

### PR DESCRIPTION
**Pull request by shrouti1507**
## Description of the change

> `content_id` and `contents` array is not mandatory. We were throwing error for this. This got edited in cloud mode

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/594)

### CODE REVIEW

File: integrations/FacebookPixel/browser.js [hunk 1]
```diff
@@ -292,35 +281,32 @@ class FacebookPixel {
...
       }
+      const productInfo = {
+        content_ids: contentIds,
+        content_type: this.getContentType(rudderElement, ["product"]),
+
+        content_name: prodName || "",
+        content_category: category || "",
+        currency: currVal,
+        value: useValue ? this.formatRevenue(value) : this.formatRevenue(price),
+        contents,
```
Comment (ItsSudip) [inline integrations/FacebookPixel/browser.js:303]: Here also we are sending empty contents array.

File: integrations/FacebookPixel/browser.js [hunk 2]
```diff
@@ -188,25 +184,22 @@ class FacebookPixel {
...
+      } else if (category) {
+        contentIds.push(category);
         contents.push({
-          id: category || "",
+          id: category,
           quantity: 1,
         });
         contentType = ["product_group"];
       }
+
```
[the remark this answers is not in the document]
Comment (shrouti1507) [reply integrations/FacebookPixel/browser.js:202]: yes. We can send it as empty array

File: integrations/FacebookPixel/browser.js (hunk 1 above)
Comment (shrouti1507) [reply integrations/FacebookPixel/browser.js:303]: yes. No error is thrown

**Review by ItsSudip — APPROVED** [review]

**Review by saikumarrs — APPROVED** [review]
Please break the track function into manageable smaller functions. Also, that method still has a great deal of code repetition. You've done a good job in clearing some of it.

**Pull request merged**